### ISSUE TITLE
Message Structure or Message Type??

### ISSUE THREAD

**Issue by csnewman88** [haffonist/igamt#1536]
When adding a profile component at the "message" level the user gets a list of all the conformance profiles in the IG. In the example below, the ADT^A01^ADT_A01 and ADT^A04^ADT_A01 are different Message Types but they share a Message Structure. This presentation makes it seem like the profile component will be at the Message Type level:
![image](https://cloud.githubusercontent.com/assets/17408249/24812119/7a109a84-1b8e-11e7-8917-b3e49c52d401.png)

However in the Definition activity, it just lists the Message Structure:
![image](https://cloud.githubusercontent.com/assets/17408249/24812305/28563720-1b8f-11e7-9332-830b5ec8f01b.png)

And when you apply that component to the ADT^A04^ADT_A01 Core Profile, it is valid:
![image](https://cloud.githubusercontent.com/assets/17408249/24812405/86bf5fc6-1b8f-11e7-9789-140cc6060a4d.png)

So the question is...
Should the profile component be at the Message Structure or Message Type level? I think there are pros and cons to both. I think creating it at the Message Type will be more understandable to users, but it may require duplicate components being created if you wanted to share an override between similar Message Types. Creating it at the Message Structure level would allow the component to be used more broadly, but the Component Profile creation window would need to be modified to filter out duplicate Message Structures and only display the Message Structure and not the Message Type. This could be confusing to a user if they forget that a Message Structure can be the same for different Message Types (for example, they forget that the A04 uses the ADT_A01 message structure).

**Comment by afrold** [comment]
@csnewman88  The profile component is specific to one conformance profile. We will display the title of the conformance profile instead of the message structure or message type.

**Comment by csnewman88** [comment]
Is it really specific to a conformance profile as soon as it's created? I would have thought it needed to be more general to that. If I have multiple conformance profiles using a common message type (RSP^K11 for example) and I want to apply a component profile to a  couple of them (but not all of them) will I need to make duplicate component profiles, one for each conformance profile?

**Comment by IsmailMellouli** [comment]
@csnewman88  The problem with this is if you change something a one of your conformance profiles, you can't have access to those changes. For example if you change a segment flavor you can't have access to it if there is only a message structure level.
What do you think?

**Comment by csnewman88** [comment]
That's true but in that case if there is always going to be a 1:1 relationship between the profile component and the conformance profile, why do you have to create a composite profile at all when the profile component is at the message level? I can see needed to create a composite profile when the component override is created at the segment level, but not at the message level. 

I think this is complex to think about on a Friday afternoon. I'm not sure I grasping all the implications.

If you go to displaying the name of the conformance profile (rather than the message structure/type), you'll need to do it everywhere. In the example below, I created a single profile component and added locations from two different conformance profiles using the same message type (unrealistic I know) and the Locations are indistinguishable (orange arrows).

![image](https://cloud.githubusercontent.com/assets/17408249/24817546/7c3b8602-1ba2-11e7-867c-faeb72a18d7b.png)
